projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/services/linked-cases.service.ts

```diff
@@ -0,0 +1,148 @@
+import { Injectable } from '@angular/core';
+import { forkJoin } from 'rxjs';
+import { CaseView } from '../../../../domain/case-view';
+import { Jurisdiction } from '../../../../domain/definition/jurisdiction.model';
+import { JurisdictionService, SearchService } from '../../../../services';
+import { LovRefDataModel } from '../../../../services/common-data-service/common-data-service';
+import { CaseLink, ESQueryType, LinkReason } from '../domain';
+
+@Injectable()
+export class LinkedCasesService {
+
+  private static readonly CASE_NAME_MISSING_TEXT = 'Case name missing';
+
+  public caseFieldValue = [];
+  public isLinkedCasesEventTrigger = false;
+  public caseDetails: CaseView;
+  public caseId: string;
+  public caseName: string;
+  public linkCaseReasons: LovRefDataModel[] = [];
+  public linkedCases: CaseLink[] = [];
+  public initialCaseLinks: CaseLink[] = [];
+  public editMode = false;
+  public jurisdictionsResponse: Jurisdiction[] = [];
+  public serverJurisdictionError: boolean;
+  public serverError: { id: string, message: string } = null;
+  public serverLinkedApiError: { id: string, message: string } = null;
+  public isServerReasonCodeError = false;
+  public caseJurisdictionID = null;
+
+  constructor(private readonly jurisdictionService: JurisdictionService,
+              private readonly searchService: SearchService) {
+    this.jurisdictionService.getJurisdictions().subscribe((jurisdictions) => {
+      this.jurisdictionsResponse = jurisdictions;
+    }, (error) => {
+      this.serverJurisdictionError = true;
+    });
+  }
+
+  public groupLinkedCasesByCaseType = (arrObj, key) => {
```

Comment:
@johnbenjamin-hmcts - not sure if you are able to specify a return type for this function

Reply:
@Daniel-Lam I have updated the code now.